src/packages/core/src/schema-builder.ts

```diff
@@ -0,0 +1,832 @@
+						fields[field.name] = { type: new GraphQLNonNull(ID) };
+						break;
+					default:
+						break;
+				}
+				continue;
+			}
+
+			// Let's try to resolve the GraphQL type involved here.
+			const { fieldType, metadata, isList } = getFieldTypeFromFieldMetadata(field);
+
+			if (isEntityMetadata(metadata)) {
+				// This if is separate to stop us cascading down to the scalar branch for entities that
+				if (
+					!metadata.apiOptions?.excludeFromBuiltInOperations &&
+					!metadata.apiOptions?.excludeFromBuiltInWriteOperations
+				) {
+					let type: GraphQLInputType = createOrUpdateTypeForEntity(metadata);
+
+					if (isList) {
+						// If it's a many relationship we need to wrap in non-null and list.
+						type = new GraphQLList(new GraphQLNonNull(type));
+					}
+
+					fields[field.name] = { type };
+				}
+			} else if (isEnumMetadata(metadata)) {
+				fields[field.name] = { type: graphQLTypeForEnum(metadata) };
+			} else {
+				fields[field.name] = { type: graphQLScalarForTypeScriptType(fieldType) };
+			}
+
+			// If it's not a nullable field and has no default then we should wrap it now in a not null.
+			if (
+				input === 'insert' &&
+				!field.nullable &&
+				typeof field.defaultValue === 'undefined' &&
+				field.relationshipInfo === undefined
+			) {
+				fields[field.name] = { type: new GraphQLNonNull(fields[field.name].type) };
+			}
+		}
+
+		return fields;
+	};
+
+const insertTypeForEntity = (entity: EntityMetadata<any, any>) => {
+	let insertType = insertTypes.get(entity.name);
+
+	if (!insertType) {
+		insertType = new GraphQLInputObjectType({
+			name: `${entity.name}InsertInput`,
+			description: `Data needed to create ${entity.plural}.`,
+			fields: generateGraphQLInputFieldsForEntity(entity, 'insert'),
+		});
+
+		insertTypes.set(entity.name, insertType);
+	}
+
+	return insertType;
+};
+
+const createOrUpdateTypeForEntity = (entity: EntityMetadata<any, any>) => {
+	let createOrUpdateType = createOrUpdateTypes.get(entity.name);
+
+	if (!createOrUpdateType) {
+		createOrUpdateType = new GraphQLInputObjectType({
+			name: `${entity.name}CreateOrUpdateInput`,
+			description: `Data needed to create or update ${entity.plural}. If an ID is passed, this is an update, otherwise it's an insert.`,
+			fields: generateGraphQLInputFieldsForEntity(entity, 'createOrUpdate'),
+		});
+
+		createOrUpdateTypes.set(entity.name, createOrUpdateType);
+	}
+
+	return createOrUpdateType;
+};
+
+const updateTypeForEntity = (entity: EntityMetadata<any, any>) => {
+	let updateType = updateTypes.get(entity.name);
+
+	if (!updateType) {
+		updateType = new GraphQLInputObjectType({
+			name: `${entity.name}UpdateInput`,
+			description: `Data needed to update ${entity.plural}. An ID must be passed.`,
+			fields: generateGraphQLInputFieldsForEntity(entity, 'update'),
+		});
+
+		updateTypes.set(entity.name, updateType);
+	}
+
+	return updateType;
+};
+
+export interface SchemaBuilderOptions {
+	authChecker?: AuthChecker<any, any>;
```

Comment:
I think this is a hangover from TypeGraphQL. Is this actually getting called anywhere now?